src/pages/posts/[id].tsx

```diff
@@ -5,35 +5,58 @@ import utilStyles from '../../styles/utils.module.css';
 
 //URLの動的に変わる部分のパスを返し、pathsに含まれないパスは404ページを表示
 export async function getStaticPaths() {
-  const paths = getAllPostIds();
+  const ids = await getAllPostIds();
+
+  const paths = ids.map((id) => {
+    return {
+      params: {
+        id: id,
+      },
+    };
+  });
```

Comment:
[IMO]
冗長なのでまとめちゃいましょ〜

```suggestion
  const paths = ids.map((id) => { params: { id }});
```